### PR TITLE
Docker credentials resource is added.

### PR DESCRIPTION
**Pull request by tahir-duplo**
- Docker credentials resource is added.
- Delete is implemented using update API by making all fields empty, We don't have a delete API.
- DNS name is added in Duplo alb resource.
- 90 seconds wait added for elasticache delete.
- availability_zones added for the plan data source.

### CODE REVIEW

File: duplocloud/resource_duplo_docker_credentials.go [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+			Description:  "The GUID of the tenant that the docker credentials will be created in.",
+			Type:         schema.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.IsUUID,
+		},
+		"user_name": {
+			Type:     schema.TypeString,
+			Required: true,
+			ForceNew: true,
```
Comment (joek-duplo) [inline duplocloud/resource_duplo_docker_credentials.go:26]: we should not `ForceNew` here.

File: duplocloud/resource_duplo_docker_credentials.go [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+		},
+		"password": {
+			Type:      schema.TypeString,
+			Required:  true,
+			Sensitive: true,
+			ForceNew:  true,
```
Comment (joek-duplo) [inline duplocloud/resource_duplo_docker_credentials.go:32]: we should not `ForceNew` here.

File: duplocloud/resource_duplo_docker_credentials.go [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+		},
+		"email": {
+			Type:     schema.TypeString,
+			Required: true,
+			ForceNew: true,
```
Comment (joek-duplo) [inline duplocloud/resource_duplo_docker_credentials.go:37]: we should not `ForceNew` here.

File: duplocloud/resource_duplo_docker_credentials.go [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+		},
+		"registry": {
+			Type:     schema.TypeString,
```
Comment (joek-duplo) [inline duplocloud/resource_duplo_docker_credentials.go:40]: we should also have `Computed = true` here

**Review by joek-duplo — CHANGES_REQUESTED** [review]

File: duplocloud/resource_duplo_docker_credentials.go (hunk 4 above)
Comment (tahir-duplo) [reply duplocloud/resource_duplo_docker_credentials.go:40]: done

File: duplocloud/resource_duplo_docker_credentials.go (hunk 3 above)
Comment (tahir-duplo) [reply duplocloud/resource_duplo_docker_credentials.go:37]: done

File: duplocloud/resource_duplo_docker_credentials.go (hunk 2 above)
Comment (tahir-duplo) [reply duplocloud/resource_duplo_docker_credentials.go:32]: done

File: duplocloud/resource_duplo_docker_credentials.go (hunk 1 above)
Comment (tahir-duplo) [reply duplocloud/resource_duplo_docker_credentials.go:26]: done

**Review by joek-duplo — APPROVED** [review]

File: duplocloud/resource_duplo_docker_credentials.go (hunk 4 above)
Comment (joek-duplo) [reply duplocloud/resource_duplo_docker_credentials.go:37]: @tahir-duplo  - it looks like `Computed: true` is still missing?

**Review by joek-duplo — CHANGES_REQUESTED** [review]

**Review by joek-duplo — APPROVED** [review]

**Pull request merged**